starknet-core/src/crypto.rs

```diff
@@ -0,0 +1,83 @@
+use ethereum_types::{H256, U256};
+use starknet_crypto::{pedersen_hash, FieldElement};
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum PedersenHashError {
+    #[error("data must not be empty")]
+    EmptyData,
+    #[error("element out of range: {0}")]
+    ElementOutOfRange(U256),
+}
+
+pub fn compute_hash_on_elements(data: &[U256]) -> Result<H256, PedersenHashError> {
+    if data.is_empty() {
+        return Err(PedersenHashError::EmptyData);
+    }
+
+    // unwrap() is safe here as it'll always succeed
+    let mut current_hash = FieldElement::from_bytes_be([0u8; 32]).unwrap();
+
+    for item in data.iter() {
+        current_hash = pedersen_hash(
+            &current_hash,
+            &u256_to_field_element(item).ok_or(PedersenHashError::ElementOutOfRange(*item))?,
+        );
+    }
+
+    let data_len = U256::from(data.len());
+    current_hash = pedersen_hash(
+        &current_hash,
+        &u256_to_field_element(&data_len).ok_or(PedersenHashError::ElementOutOfRange(data_len))?,
+    );
+
+    Ok(H256::from_slice(&current_hash.to_bytes_le()[..]))
+}
+
+fn u256_to_field_element(num: &U256) -> Option<FieldElement> {
```

Comment:
Just curious here - wouldn't it be cleaner (which is subjective) if this function lived in `field_element.rs` in `starknet-crypto`?

Reply:
`starknet-crypto` isn't aware of `U256`. The `U256` in `starknet-crypto` is `crypto_bigint::U256` instead of the `primitive_types::U256` here, despite the same name lol.

Actually we should be getting rid of `U256` and `H256` altogether... as suggested in #13.